src/driver/mongodb/typings.ts

```diff
@@ -4487,7 +4487,7 @@ export declare class MongoClient extends TypedEventEmitter<MongoClientEvents> {
      *
      * @see docs.mongodb.org/manual/reference/connection-string/
      */
-    connect(): Promise<this>
+    connect(url: string, options: any): Promise<this>
```

Comment:
We pass the url and options in the constructor.
This would also lead to a breaking change of the API